src/dbi.c

```diff
@@ -407,7 +414,7 @@ static int cdbi_config_add_database (oconfig_item_t *ci) /* {{{ */
       plugin_register_complex_read (/* group = */ NULL,
           /* name = */ name ? name : db->name,
           /* callback = */ cdbi_read_database,
-          /* interval = */ 0,
+          /* interval = */ (db->interval > 0) ? &interval : NULL,
```

Comment:
Shouldn't this read as:
`(db->interval > 0) ? &interval : 0,`
instead ? ie: keep the default 0.